torch/_inductor/codegen/wrapper.py

```diff
@@ -1004,10 +1005,16 @@ def write_wrapper_decl(self):
                 isinstance(v, torch.Tensor) for v in list(V.graph.constants.values())
             ), "Expect all constants to be Tensor"
             for idx, constants_key in enumerate(V.graph.constants.keys()):
-                constants_idx = inputs_len + idx
-                self.prefix.writeline(
-                    f"at::Tensor {constants_key} = args[{constants_idx}];"
-                )
+                if V.graph.aot_mode:
+                    self.prefix.writeline(
+                        f"""at::Tensor {constants_key} = constants_->at("{constants_key}");"""
```

Comment:
Is this called in the Model's ```run_impl``` method? If so, I am wondering if we could pre-set the key-to-index map so that we can have ```O(1)``` access to fetch a constant tensor, if it's not too hard. My worry is that the runtime overhead might be high if we had a large number of constants.

Reply:
This is an `unordered_map`, so "in theory" it should be a O(1) access already. Key-to-index shouldn't matter much. as for not using `shared_ptr` this would be plenty of work I guess...

Reply:
This itself is an unordered_map, so we should have O(1) access to fetch the constant tensor. It is, possible to use a vector/list to hold the correct index. But if someday we want users to provide a mapping of constant name -> tensor, that's not possible for us to correctly render the index at compile time.